tock/api/renderers.py

```diff
@@ -16,3 +21,38 @@ def render(self, data, media_type=None, renderer_context=None):
         if not self.headers and len(data) > 0:
             self.headers = data[0].keys()
         return super(PaginatedCSVRenderer, self).render(data, media_type, renderer_context)
+
+def stream_csv(queryset, serializer):
+    """
+    Stream data as CSV, given an interable queryset and a DRF
+    Serializer instance with a .fields dict and .to_representation()
+    method.
+    """
+    rows = map(serializer.to_representation, queryset)
+    fields = list(serializer.fields.keys())
+    response = StreamingHttpResponse(generate_csv(rows, fields), content_type='text/csv')
+    return response
+
+class Echo(object):
+    """
+    A pseudo-buffer, see:
+    <https://docs.djangoproject.com/en/1.8/howto/outputting-csv/#streaming-large-csv-files>
+    """
+    def write(self, value):
+        return value
```

Comment:
This isn't a problem, but we don't *have* to return `value` here, right? AFAIK, the only purpose of this class is to present a file-like interface, but not actually to do any work. It's basically a hack to deal with the fact that the `csv` module only wants to work with file-like objects.